### PR TITLE
Split up compiled html for each doc object

### PR DESCRIPTION
**Pull request by vanessayuenn**
The purpose of this PR is to split up the `html` prop of each doc object into smaller chunks, allowing more granular control over how it's displayed on electronjs.org. It does so by first parsing the markdown document into "sections", which are defined by _any heading, and its following content until the next heading_. Then each of the section is compiled to html in the same way as before.

Considerations:
- This PR will change the `html` prop of each doc object from a string of the entire compiled html, to an array containing compiled html of smaller sections, so it will be a _breaking change_.
- We could consider retaining the original format of `html`, but returning the new html sections as a new property for now until the website is caught up. So we can merge new translations without breaking the site. 
- Right now to access the html of a doc section you have to get `i18n.docs[req.language][req.path].html[index].html`, which imo is too many confusing `html`.

### CODE REVIEW

File: script/build-module.js [hunk 1]
```diff
@@ -65,46 +65,53 @@ async function parseFile (file) {
...
-
-    const parsed = URL.parse(newSrc)
-    parsed.path = path.normalize(parsed.path)
-
-    $(el).attr('src', URL.format(parsed))
-  })
-
-  file.html = $('body').html()
+  file.html = await Promise.all(
+    splitMd(file.markdown).map(async (section) => {
```
Comment (zeke) [inline script/build-module.js]: I think changing this to a standalone named function would make this code a bit easier to follow, e.g.

```js
file.html = await Promise.all(splitMd(file.markdown).map(processSection))
```

File: script/build-module.js [hunk 2]
```diff
@@ -117,6 +124,34 @@ async function parseFile (file) {
...
+  const sections = []
+  let section = { name: null, body: [] }
+  let inCodeBlock = false
+  const isHeading = (line) => (!inCodeBlock && line.trim().startsWith('#'))
+
+  md.split('\n').forEach((curr, i, lines) => {
+    if (curr.startsWith('```')) {
+      inCodeBlock = !inCodeBlock
+    }
+    if (isHeading(curr)) {
```
Comment (zeke) [inline script/build-module.js]: I would recommend using [slugg](https://ghub.io/slugg) for this. We're already using it in `electron-apps` and elsewhere.

Comment (vanessayuenn) [reply script/build-module.js]: Cool! I didn't know about this!

**Comment by vanessayuenn** [comment]
@zeke this PR is ready for another 👀 . I switched out `slugg` for [`github-slugger`](https://www.npmjs.com/package/github-slugger) instead as that's the one already used in `remark-slug` so it's more consistent, and is more appropriate for our uses (e.g. we don't need unicode stripped from class names).

When this is ready, we will need to do a major/breaking release. How is that done with the current `sematic-release` setup?

**Comment by zeke** [comment]
@vanessayuenn I added a line to the module-building script that stitches the sections back into HTML. This maintains module compatibility and lets us avoid all the section-stitching we were doing in the tests before to test the HTML. This also doesn't add any significant processing overhead, as we're now only parsing markdown to HTML once.

I also added a test to verify that all the sections are present and have valid content, though that test is currently breaking:

```
AssertionError: en-US /docs/README has a section without a name: expected undefined to be a string
```

README is kind of an edge case docs-wise. We've got that title bug https://github.com/electron/electron-i18n/issues/203, and I'm thinking the `/docs/README` URL on the site should be deprecated/redirected to `/docs` once we implement https://github.com/electron/electron/issues/11416 -- so... maybe we should just disregard README because it's "different"?

**Comment by vanessayuenn** [comment]
@zeke I am cool with keeping the full `html` property for now just to be safe. But I think once we are sure things are working as expected on the website side, we should take it out to shave the size of the whole i18n object in half. 

>I also added a test to verify that all the sections are present and have valid content

☝️ this is nice!
>README is kind of an edge case docs-wise. We've got that title bug #203, and I'm thinking the /docs/README URL on the site should be deprecated/redirected to /docs once we implement electron/electron#11416 -- so... maybe we should just disregard README because it's "different"?

I think we should have the README file ignored in the build process, at least for now, until we give the README a big overhaul and implement what you suggested [here](https://github.com/electron/electron-i18n/issues/203#issuecomment-353665084). I think we can skip the redirecting though; I did a quick search and nothing points to that URL. The only way to access `/docs/README` is if it comes up as a search result (which is rather frequent), and that won't be an issue if README is not included in `docs` at all. Can we use the ignore pattern `<!-- i18n-ignore -->` to tell the build process to ignore all the README's?

**Comment by vanessayuenn** [comment]
🤔 `/docs/development/updgrading-chrome` also has the same issue. @zeke, is there any consequence of just removing [this file](https://github.com/electron/electron/blob/master/docs/development/updgrading-chrome.md) from the electron repo?

**Comment by zeke** [comment]
> shave the size of the whole i18n object in half.

We should probably remove the `markdown` string too.

> any consequence of just removing this file from the electron repo?

We kept that file around to prevent 404s, but it looks like the only one of its kind. I'd be okay with removing it. There are probably not many sites linking to that specific URL...

**Comment by vanessayuenn** [comment]
Hrm looks like the capitalization of `Feat` in the commit message did _not_ trigger a minor release. 😓

**Comment by zeke** [comment]
@vanessayuenn you could just merge https://github.com/electron/electron-i18n/pull/238 with a semantic message and this PR will get rolled into the next release.
